test/functional/tests/upgrade_in_flight/test_upgrade_basic.py

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright(c) 2019-2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+import time
+
+from api.cas import casadm, version, git, installer, upgrade_in_flight
+from api.cas.cache_config import CacheLineSize
+from api.cas.casadm_parser import (
+    get_caches,
+    get_cas_cache_version,
+    get_cas_disk_version,
+    get_casadm_version,
+    get_cores,
+    get_statistics,
+)
+from api.cas.statistics import CacheStats, CoreStats
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.output import CmdException
+from upgrage_test_utils import *
+
+
+@pytest.mark.require_plugin("power_control")
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize(
+    "cache_line_size",
+    [CacheLineSize.LINE_4KiB, CacheLineSize.LINE_16KiB, CacheLineSize.LINE_64KiB],
+)
+@pytest.mark.parametrize(
+    "cas_version_to_upgrade", [version.CasVersion("v20.1"), version.CasVersion("v20.3")]
+)
+def test_cas_upgrade_during_io(cache_line_size, cas_version_to_upgrade):
+    """
+    title: Upgrade CAS during IO
+    description: |
+      Verify CAS ability to upgrade in flight during IO.
+    pass_criteria:
+      - casadm and cas_cache version is updated
+      - none of the CAS files (udev rules, casctl utils) are not changed
+      - cas_disk version not changed until reboot
+      - init config not changed
+      - No system crash occured
+      - CAS is working in the same configuration as before upgrade
+      - Data integrity is not broken
+    """
+    with TestRun.step("Collect all versions and install older CAS"):
+        original_commit = git.get_current_commit_hash(from_dut=True)
+        original_casadm_version = get_casadm_version()
+        original_cas_cache_version = get_cas_cache_version()
```

Comment:
Shouldn't this be below set-up like in next test?

Reply:
No, because in this test I expect CAS to be upgraded to the same version as was installed before running test and in the second one CAS version should not be changed